### ISSUE TITLE
Job Description overview

### ISSUE THREAD

**Issue by Oanh-and-only-Oanh** [topcoder-platform/community-app#4659]
build front-end to allow members to input data for job descriptions/bullet points

**Comment by Oanh-and-only-Oanh** [comment]
@Dara-K , can you help do a quick mock up of front-end design of what this would look like to hand over to dev?

**Comment by Oanh-and-only-Oanh** [comment]
https://marvelapp.com/prototype/797cb4i/screen/71547769

**Comment by skyhit** [comment]
@Oanh-and-only-Oanh how are we going to save the data, through APIs? are they ready?

**Comment by Oanh-and-only-Oanh** [comment]
The api is not ready. Working on this now.

**Comment by Oanh-and-only-Oanh** [comment]
@urwithat, can you help add the ability to store this new data field into the member API?

**Comment by urwithat** [comment]
@Oanh-and-only-Oanh - we have the work traits endpoint exposed already. The new attributes can be passed to the same endpoint.
An issue might arise while reading old records in our data store that don't have jobDescription, achievements & technologies attributes, that can be easily handled on the front end.

Sample CURL for fetching:
curl --location --request GET 'https://api.topcoder-dev.com/v5/members/upbeat/traits?traitIds=work' \
--header 'Authorization: Bearer token'

Sample CURL for creating:
curl --location --request POST 'https://api.topcoder-dev.com/v5/members/upbeat/traits' \
--header 'Authorization: Bearer token' \
--header 'Content-Type: application/json' \
--data-raw '[
    {
        "categoryName": "Work",
        "traitId": "work",
        "traits": {
            "data": [
                {
                    "company": "Company 1",
                    "industry": "Industry 1",
                    "position": "Position 1",
                    "cityTown": "City Town 1",
                    "timePeriodFrom": "2017-11-16T06:30:00.000Z",
                    "timePeriodTo": "2020-05-01T12:00:00.000Z",
                    "working": false,
                    "jobDescription": "jobDescription 1",
                    "achievements": "achievements 1",
                    "technologies": "technologies 1"
                }
            ]
        }
    }
]'

**Comment by skyhit** [comment]
@Oanh-and-only-Oanh Can you provide the exported svg icons for the editor toolbar's buttons?

**Comment by Oanh-and-only-Oanh** [comment]
@Dara-K, can you provide requested svg icons?

**Comment by luizrrodrigues** [comment]
@skyhit Added feedback to both PRs.

**Comment by luizrrodrigues** [comment]
@skyhit **community-app** PR looks good.

**topcoder-react-lib** PR not fixed yet.
https://github.com/topcoder-platform/topcoder-react-lib/pull/261

**Comment by luizrrodrigues** [comment]
@skyhit Any status on this one?

**Comment by skyhit** [comment]
@luizrrodrigues @Oanh-and-only-Oanh I asked for a fix now, will keep you update

**Comment by skyhit** [comment]
@luizrrodrigues @Oanh-and-only-Oanh fix for topcoder-platform/topcoder-react-lib#261 is ready

**Comment by SathyaJayabal** [comment]
1. Text "Job Description" displaying on top of the date picker
<img width="1440" alt="Screenshot 2020-10-13 at 2 31 00 PM" src="https://user-images.githubusercontent.com/58783823/95839665-e67e7200-0d60-11eb-88b4-b42add5ba5db.png">
2. UX Suggestion: User should be able to type ahead and select technologies. It is very time consuming to scroll through the entire list to select your technology
3. "Failed to update User Trait" error on trying to add new Work.
<img width="1440" alt="Screenshot 2020-10-13 at 2 30 08 PM" src="https://user-images.githubusercontent.com/58783823/95840320-9fdd4780-0d61-11eb-8bbf-969fdf6da87e.png">

**Comment by skyhit** [comment]
@Oanh-and-only-Oanh @SathyaJayabal a new PR is created please review - https://github.com/topcoder-platform/community-app/pull/5139

**Comment by SathyaJayabal** [comment]
@skyhit @luizrrodrigues , we are getting a blank page on accessing profile  settings 
https://staging-community-app.topcoder.com/settings/profile
<img width="1440" alt="Screenshot 2020-11-04 at 12 17 52 PM" src="https://user-images.githubusercontent.com/58783823/98078330-e3762d80-1e97-11eb-83ba-955592904916.png">

**Comment by luizrrodrigues** [comment]
Thanks @SathyaJayabal 

@skyhit Checking about PRs.

We can ignore previous **community-app** PR: https://github.com/topcoder-platform/community-app/pull/5011 right?

The new one, https://github.com/topcoder-platform/community-app/pull/5139 need to point to previous **topcoder-react-lib** PR https://github.com/topcoder-platform/topcoder-react-lib/pull/261

Right?

**Comment by skyhit** [comment]
@luizrrodrigues how to do that? should we merge that PR first?

**Comment by luizrrodrigues** [comment]
@skyhit just confirm if need both PRs, then I'll create a test release and deploy.

**Comment by luizrrodrigues** [comment]
@skyhit looks like **community-app** and **topcoder-react-lib** PR was not synced, then I merge develop into both branches. 

Deploying to Dev env.

**Comment by SathyaJayabal** [comment]
@skyhit, Technology list is not loading.
<img width="1440" alt="Screenshot 2020-11-09 at 4 11 10 PM" src="https://user-images.githubusercontent.com/58783823/98531972-13ae3980-22a7-11eb-93dd-9ce1a440fd65.png">

**Comment by luizrrodrigues** [comment]
@skyhit I have fixed this in **topcoder-react-lib**, was calling API v3 to `/technologies` instead v4.
https://github.com/topcoder-platform/topcoder-react-lib/pull/261/commits/8ed33653ff83467a23d49a9b7cc2519205f2a351

**Comment by luizrrodrigues** [comment]
@SathyaJayabal Fixed and ready for QA.

**Comment by SathyaJayabal** [comment]
@luizrrodrigues @skyhit , verified.
<img width="1440" alt="Screenshot 2020-11-09 at 9 27 09 PM" src="https://user-images.githubusercontent.com/58783823/98564419-5fc2a380-22d2-11eb-9bc2-c7971c7a6687.png">

**Comment by SathyaJayabal** [comment]
verified
<img width="1440" alt="Screenshot 2020-11-10 at 1 42 30 PM" src="https://user-images.githubusercontent.com/58783823/98647242-9a701e80-235a-11eb-8719-555ec4524242.png">

**Comment by SathyaJayabal** [comment]
@luizrrodrigues , looks like in prod api call is to 
https://api.topcoder.com/v3/members/picachui/traits  and staging is
https://api.topcoder.com/v5/members/picachui/traits

users profile details are not getting displayed on staging
<img width="1440" alt="Screenshot 2020-11-10 at 3 12 33 PM" src="https://user-images.githubusercontent.com/58783823/98656870-32740500-2367-11eb-9adf-0534e53befa6.png">
<img width="1440" alt="Screenshot 2020-11-10 at 3 12 37 PM" src="https://user-images.githubusercontent.com/58783823/98656879-3738b900-2367-11eb-921d-17db7e3d6a7b.png">

**Comment by luizrrodrigues** [comment]
@SathyaJayabal Here V5 Traits Loading, but data no consistence: 
![image](https://user-images.githubusercontent.com/7029641/98663764-42202900-2328-11eb-9503-f675b916a000.png)

**Comment by skyhit** [comment]
@luizrrodrigues @SathyaJayabal so we should adapt for V5 API? and does V5 API support this?

**Comment by SathyaJayabal** [comment]
@skyhit , yes. @luizrrodrigues this may need api updates too? Looks like data returned by the v3 and v5 apis for the same user is different. @skyhit you will need to coordinate with @urwithat for api  related updates.

**Comment by Oanh-and-only-Oanh** [comment]
@urwithat @skyhit, what's the status on this?

**Comment by luizrrodrigues** [comment]
@skyhit I think we need just v3 to `traits` for now, like existing in Prod.

**Comment by urwithat** [comment]
@skyhit - Use the member v5 microservices.
@SathyaJayabal - The traits data stored for your member seem to have an issue, have you done any manual changes in any data store or executed any create / update member curl query.
I've synced your member's data, its working now. Please check and close this issue.

**Comment by SathyaJayabal** [comment]
@urwithat , I did not do anything to my data other than updation through the community app. @luizrrodrigues also had the same issues. Is it possible to check for some other users too.

**Comment by urwithat** [comment]
@SathyaJayabal - Have checked for a couple of other members, seems to be fine. Let me know if you find any other members with such an issue.

**Comment by SathyaJayabal** [comment]
@luizrrodrigues , let deploy this and test, better to use staging.

**Comment by luizrrodrigues** [comment]
@SathyaJayabal @urwithat Deployed to Stag env.

![image](https://user-images.githubusercontent.com/7029641/100206447-1a62c080-2ee5-11eb-99df-a47013ce7579.png)

**Comment by SathyaJayabal** [comment]
@luizrrodrigues @skyhit , i don't see an error for the traits api call now. But my profile info for basic details is not getting populated. Maybe because the data format is diff between v3 and v5. 
<img width="1437" alt="Screenshot 2020-11-25 at 3 24 24 PM" src="https://user-images.githubusercontent.com/58783823/100211957-c8e12280-2f32-11eb-8666-f4f0e8123334.png">

**Comment by luizrrodrigues** [comment]
@SathyaJayabal Is same issue in screen above, v5 missing some data.

**Comment by luizrrodrigues** [comment]
@urwithat @skyhit Any update on this one? Please let me know if I can use Stag env. (I need to deploy a hotfix to QA)

**Comment by luizrrodrigues** [comment]
We're moving this issue to Backlog for now.
